### PR TITLE
Add EIP: Uncouple blob count between CL and EL

### PR DESCRIPTION
**Pull request by ralexstokes**
**ATTENTION: ERC-RELATED PULL REQUESTS NOW OCCUR IN [ETHEREUM/ERCS](https://github.com/ethereum/ercs)**

--

When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-7733.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+eip: TBD
```
Comment (abcoathup) [inline EIPS/eip-7733.md:2]:
```suggestion
eip: 7742
```
Assigning next sequential EIP/ERC/RIP number.

Please also update the filename.

File: EIPS/eip-7733.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+1) Gain more flexibility over the value, rather than the static `TARGET == MAX // 2` relation in EIP-4844.
+2) Uncouple development and deployment of the CL and EL layers in the event it is desirable to change the blob target value.
+
+## Specification
+
+| constants                    | value                                        |
+|---                           |---                                           |
+| `FORK_TIMESTAMP`             | TBD                                          |
+
+### Background
```
Comment (mkalinin) [inline EIPS/eip-7733.md:35]: IMHO, this should be moved to Motivation or partially to Rationale

File: EIPS/eip-7733.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+#### Target amount of blobs per block
+
+The target is currently specified as a fixed value in relation to the blob count. The Ethereum community intends to increase
+the blob parameters as part of its scaling strategy and the ability to have a more flexible target value in relation to
+the blob max is desirable to reduce rigidity in this protocol parameter.
+
+Even if the EL keeps a fixed target value based on the max, removing the max implies the EL would not know what the target
+value should be. To address this lack of information, this EIP proposes the CL sends the current target value to the EL
+with each provided payload over the Engine API. The EL block header will also need to be extended with this target value
```
Comment (mkalinin) [inline EIPS/eip-7733.md:63]: AFAICS, `target_blob_gas_per_block` is used to compute `excess_blob_gas`. Would it be possible to send the target as a part of `PayloadAttributes` so the EL can compute the excess and make CL validate that the excess is computed correctly given the target and parent’s excess?

Comment (ralexstokes) [reply EIPS/eip-7733.md]: the CL part of this change is here: https://github.com/ethereum/consensus-specs/pull/3800

it hasn't been updated yet but yes, the CL will send the target value in the `PayloadAttributes` so the EL can build a valid block

Comment (mkalinin) [reply EIPS/eip-7733.md]: Sorry, i wasn’t clear enough on the CL validation part, it seems to be possible to me to not include `target_blob_gas_per_block` into EL block by doing the following:
1) Send this parameter to payload build process so EL can correctly compute `excess_blob_gas`
2) Make CL validate that the computation was done correctly by utilising a simple math: `assert payload.excess_blob_gas == min(0, state.latest_execution_payload_header.excess_blob_gas + target_blob_gas_per_block - blob_gas_used)`

Comment (ralexstokes) [reply EIPS/eip-7733.md]: would you still commit the `target_blob_gas` in the EL header?

it seems to break the ability for the EL to independently verify the `excess_blob_gas` (and by extension the blob base fee) if it doesn't have access to the target value

if we keep the header extension then your change is just moving this check from the EL to the CL, which is fine in principle although I could argue violates the separation of concerns of "blob (max) limit check on the CL for DoS reasons" and "blob fee accounting on the EL"

Comment (mkalinin) [reply EIPS/eip-7733.md]: > it seems to break the ability for the EL to independently verify the `excess_blob_gas` (and by extension the blob base fee) if it doesn't have access to the target value

I’d argue that EL doesn’t need to have this ability as long as CL is responsible for verifying the `excess_blob_gas` computation. If we put it this way then we have no change to the EL block structure with a bit of cross-layer logic duplication: EL and CL will have to compute `excess_blob_gas`. But I think this duplication is fine as the `excess_blob_gas` formula is unlikely to change.

**Pull request merged**